backend/src/rules/abilities.clj

```diff
@@ -6,13 +6,34 @@
   (fn    
 ;   "Alters a cards' power, by adding the a defined value to it.
 ;   This power may be negative, resulting in a decrease"
-    ([game-state target]
+    ([game-state play]
      (update-in
        game-state
-       [:cards target :power]
+       [:cards (:target play) :power]
        #(+ % increase)))
      ([]
       {:description (if (neg? increase)
                       (str "Weaken " (- increase))
                       (str "Enhance " increase))
        :target 1})))
+
+(defn type-add-power
+  "Creates the ability function"
+  [row-type increase]
+  (fn
+;    "When played in a row-type row,
+;    alters a its power, by adding the a defined value to it.
+;    This power may be negative, resulting in a decrease"
+    ([game-state play]
+     (if (= (get-in game-state [:rows (:row-id play) :type])
+            row-type)
+       (update-in
+         game-state
+         [:cards (:card-id play) :power]
+         #(+ % increase))
+
+       game-state))
+    ([]
```

Comment:
I have the enormous suspicion that calling the function with no arguments to get a description is a bad idea... but I can't think of a better alternative.

Maybe to make it more reasonable we can have a function `(describe ability)` which simply calls `ability` with no parameters? I don't know, something seems very off to me.

Reply:
Oh wait, I have an idea. The function should generate both a function and a description everytime, so an ability has essentially `{:function (fn ...) :description "This is the description")`.

This separates the concerns cleanly instead of having a function return radically different things dependent on the number of arguments.

Reply:
I really like it.
Therefore cards would have `:ability some-function :description "ability description"` and, when needed, `:target 1`.

Is that right?
I'll do that change when able.

Reply:
Thanks!